prql-compiler/src/sql/gen_expr.rs

```diff
@@ -514,59 +484,66 @@ pub(super) fn translate_select_item(cid: CId, ctx: &mut Context) -> Result<Selec
     Ok(SelectItem::UnnamedExpr(expr))
 }
 
-fn try_into_is_null(
-    op: &BinOp,
-    a: &Expr,
-    b: &Expr,
-    ctx: &mut Context,
-) -> Result<Option<sql_ast::Expr>> {
-    if matches!(op, BinOp::Eq) || matches!(op, BinOp::Ne) {
-        let expr = if matches!(a.kind, ExprKind::Literal(Literal::Null)) {
-            b.kind.clone()
-        } else if matches!(b.kind, ExprKind::Literal(Literal::Null)) {
-            a.kind.clone()
-        } else {
-            return Ok(None);
-        };
+fn try_into_is_null(expr: Expr, ctx: &mut Context) -> Result<Result<sql_ast::Expr, Expr>> {
```

Comment:
Is it worth adding a note why we have wrapped `Result`s? What does it mean to have an error on the inner vs the outer?

Reply:
Yes it is, will do.